torch/csrc/distributed/rpc/rref_context.cpp

```diff
@@ -37,33 +38,117 @@ const std::shared_ptr<RpcAgent>& RRefContext::agent() const {
   return agent_;
 }
 
-void RRefContext::addFork(at::IValue&& value) {
+RRefForkData RRefContext::forkTo(
+    const std::shared_ptr<RRef>& rref, worker_id_t forkDst) {
+  auto forkRequest = rref->fork();
+  if (rref->owner() != forkDst) {
+    // if fork destination if not owner, the forked UserRRef needs to be tracked
+    // properly
+    if (rref->isOwner()) {
+      // fork from owner
+      acceptUserRRef(forkRequest.rrefId_, forkRequest.forkId_, forkDst);
+    } else {
+      // fork from user, rref cannot be destructed until the fork request is
+      // accepted by the owner
+      {
+        std::lock_guard<std::mutex> lock(mutex_);
+        pendingForkRequests_[forkRequest.forkId_] = rref;
+      }
+      // notify owner
+      agent_->send(
```

Comment:
traceability: here we're throwing away the result of this send. This means no one will be notified of any exception that happens during the execution of this function. It will make debugging really hard. We should at least install a callback on error of this send and at the very least print an error message if the notification doesn't go through.